backend/src/main/java/com/redhat/cloud/notifications/routers/sources/SourcesService.java

```diff
@@ -0,0 +1,84 @@
+package com.redhat.cloud.notifications.routers.sources;
+
+import io.quarkus.rest.client.reactive.ClientExceptionMapper;
+import org.eclipse.microprofile.rest.client.annotation.RegisterClientHeaders;
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.PATCH;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.Response;
+
+/**
+ * <p>REST Client for the Sources API. The OpenAPI spec is available at:</p>
+ *
+ * <ul>
+ *  <li><a href="https://console.stage.redhat.com/api/sources/v3.1/openapi.json">OpenApi v3.1 in the stage environment.</a></li>
+ *  <li><a href="https://console.redhat.com/api/sources/v3.1/openapi.json">OpenApi v3.1 in the production environment.</a></li>
+ *  <li><a href="https://github.com/RedHatInsights/sources-api-go/blob/main/public/openapi-3-v3.1.json">OpenApi v3.1 on GitHub.</a></li>
+ * </ul>
+ *
+ * <p>Please be aware of the following:</p>
+ *
+ * <ul>
+ *     <li>If sources is using a database backend, only the {@link Secret#password} field will get encrypted.</li>
+ *     <li>On the other hand, if sources is using the AWS Secrets Manager, then the whole secret will get encrypted.</li>
+ * </ul>
+ */
+@ApplicationScoped
+@RegisterClientHeaders(SourcesXRHIDHeaderFactory.class)
+@RegisterRestClient(configKey = "sources")
+public interface SourcesService {
+
+    /**
+     * Get a single secret from Sources. In this case we need to hit the internal endpoint —which is only available for
+     * requests coming from inside the cluster— to be able to get the password of these secrets.
+     * @param id the secret id.
+     * @return a {@link Secret} instance.
+     */
+    @GET
+    @Path("/internal/v2.0/secrets/{id}")
+    Secret getById(@PathParam("id") long id);
```

Comment:
FYI, there is a newer set of annotations available in Quarkus that is a bit more dev-friendly:
```suggestion
    Secret getById(@RestPath long id);
```

Reply:
Nice, I like that one better!